### PR TITLE
feat: support qat models in hybrid model

### PR DESCRIPTION
**Pull request by fd0r**
closes https://github.com/zama-ai/concrete-ml-internal/issues/3992

### CODE REVIEW

**Review by jfrery — DISMISSED** [review]
Nice update. Thanks!

**Review by andrei-stoian-zama — APPROVED** [review]
good idea! a unified compilation interface for both qat & ptq with automatic detection of qat.

File: src/concrete/ml/pytest/torch_models.py [hunk 1]
```diff
@@ -1387,3 +1387,25 @@ def forward(self, x):
         x = torch.relu(self.fc2(x))
         x = self.fc4(self.quant_3(x))
         return x
+
+
+class PartialQATModel(torch.nn.Module):
```
Comment (RomanBredehoft) [inline src/concrete/ml/pytest/torch_models.py:1392]: more like `HybridQatMOdel` right ?

Comment (fd0r) [reply src/concrete/ml/pytest/torch_models.py:1392]: Well it depends, the idea here is that you can have non-qat module afterwards and only compile the `sub_module` that uses qat and everything will go fine.

Comment (RomanBredehoft) [reply src/concrete/ml/pytest/torch_models.py:1392]: ah got it thanks !

**Review by RomanBredehoft — APPROVED** [review]
great thanks for this !

**Pull request merged**